### ISSUE TITLE
How to set up K3S, GlusterFS and Hetzner's cloud load balancer

### ISSUE THREAD

**Issue by GinjiJizai** [hetzneronline/community-content#662]
## Problem

Hi,

the provided guide is sadly not working entirely. Everything is fine and good explained, until the Load Balancer Part. I did exactly as described and configured the Traefik Helm as well als creating all the resources in Hetzner Cloud. If i now try to access my service via Load Balancer and HTTPS, i only get 400 errors. Has someone did connect traefik with a Hetzner Load Balancer more reacently? 

## Suggested Fix

Cross check the Guide and update it, if possible.

**Comment by svenja11** [comment]
@virtualzone Can you take a look at this?
Would you be interested in providing an update to your tutorial?

The statement in step 2.2 is now also [outdated](https://github.com/k3s-io/k3s/issues/3863#issuecomment-1424929168):
> At the time of writing, there seems to be a bug with Debian 11 which prevents K3S from using VXLAN to communicate between the nodes. See: https://github.com/k3s-io/k3s/issues/3863

**Comment by svenja11** [comment]
Hello @GinjiJizai 
I tested the tutorial and it works for me. The configuration would be a little bit different if you used an IP address instead of a domain. But in your comment you mentioned that you set up HTTPS, so I assume you do have a domain.

  * The example nginx deployment in "Step 7.2 - Deploy a webserver" includes the placeholder `<example.com>`. Did you see that you need to replace `<example.com>` with your own domain?
  * Have you double-checked if GlusterFS is setup properly and everything is working? One way to test this, is to check if the `index.html` file you created in `/mnt/gluster-k8s/webtest1` is available on all nodes. If you created this file on "master-1", this file should also automatically be available on "node-1" and "node-2".
  * Did you remember to edit your DNS entry so that your domain points at the IP address of the Load Balancer? 

It would also be interesting to know if the status of your Load Balancer targets is healthy or not. This would mean that the issue is not on the servers but actually comes from the domain resolution. If your targets are healthy, you can use an IP address instead of your domain to check if you can access the website then.

* **IP address**

  To test if the servers are available via HTTP, you need to make these changes:
  * The Load Balancer service in "Step 6.2 - Create the Load Balancer" is set up for HTTPS. To use an IP address instead of a domain, you need to replace the current Load Balancer service with this service: 
    
    | Protocol | Source port | Destination port |
    | --------- | ------------- | ------------------ |
    | http        | 80               |  80                      |
  * The example nginx deployment in "Step 7.2 - Deploy a webserver" includes the placeholder `<example.com>`. You need to replace `<example.com>` with the IP address of your Load Balancer.

If any of this helped or if you were able to resolve the issue yourself, it would be great if you could share what the issue was.

**Comment by GinjiJizai** [comment]
@svenja11 

I retested the tutorial, now i did not use the  amd machines of hetzner, but the intell ones. Seems like glusterfs wasn't setup correctly in the first place. Everything else you mentioned was correct on my side.

I also noticed that if you register a node to early under specific conditions, the kubernetes does not setup the core dns correctly, i guess that this caused the 400, because there was traffic on the loadbalancer but the service was not reached.

So everything is fine, was a problem on my side. Thank you for checking!
